### DISCUSSION TITLE
Is it possible to "inject" environment files?

### DISCUSSION THREAD

**Discussion by Drevoed** [vercel/next.js#16080] [Help]
I want to be able to build only once and then set the API endpoints depending on what lies in environment file somewhere.

Or do I really have to build a new bundle for every possible development environment?

P.S

I assume this kind of thing is only possible for pages that are SSR-only?

**Comment by jamesmosier** [comment]
Next has built in environment variable support: https://nextjs.org/docs/basic-features/environment-variables

But you'd need to build each time you changes the values in your `.env` file (for your various environments).

**Reply by Drevoed** [reply]
> Next has built in environment variable support: https://nextjs.org/docs/basic-features/environment-variables
> 
> But you'd need to build each time you changes the values in your `.env` file (for your various environments).

Did I not understand correctly or is it not even possible for server side rendered pages? Seeing as the server can grab the correct import strings at runtime based on environment variable and render that in?

**Reply by Timer** [reply]
For runtime variables used for server-rendering you can read `process.env.` like normal! No need for `.env` files.

**Reply by Drevoed** [reply]
> For runtime variables used for server-rendering you can read `process.env.` like normal! No need for `.env` files.

So seems like I will have to opt-out of static generation for pages that use the env, thanks!
